compass/ocean/tests/global_ocean/mesh/remap_mali_topography/__init__.py

```diff
@@ -0,0 +1,263 @@
+            topography is used to determine the land mask
+
+        mesh_name : str
+            The name of the MPAS mesh to include in the mapping file
+
+        mali_ais_topo : str, optional
+            Short name for the MALI dataset to use for Antarctic Ice Sheet
+            topography
+
+        ocean_includes_grounded : bool
+            Whether to include grounded cells that are below sea level in the
+            ocean domain
+        """
+        super().__init__(test_case=test_case, base_mesh_step=base_mesh_step,
+                         mesh_name=mesh_name)
+        self.mali_ais_topo = mali_ais_topo
+        self.ocean_includes_grounded = ocean_includes_grounded
+
+        self.add_output_file(filename='mali_topography_remapped.nc')
+
+    def setup(self):
+        """
+        Set up the step in the work directory, including downloading any
+        dependencies.
+        """
+        super().setup()
+
+        mali_filename = self.config.get('remap_mali_topography',
+                                        'mali_filename')
+        self.add_input_file(
+            filename='mali_topography_orig.nc',
+            target=mali_filename,
+            database='mali_topo')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        super().run()
+        self._remap_mali_topo()
+        self._combine_topo()
+
+    def _remap_mali_topo(self):
+        in_mesh_name = self.mali_ais_topo
+        in_descriptor = MpasCellMeshDescriptor(
+            fileName='mali_topography_orig.nc',
+            meshName=in_mesh_name)
+        in_descriptor.format = 'NETCDF3_64BIT'
+        in_descriptor.to_scrip('mali.scrip.nc')
+        self._partition_scrip_file('mali.scrip.nc')
+
+        out_mesh_name = self.mesh_name
+        out_descriptor = MpasCellMeshDescriptor(
+            fileName='base_mesh.nc',
+            meshName=out_mesh_name)
+        out_descriptor.format = 'NETCDF3_64BIT'
+        out_descriptor.to_scrip('mpaso.scrip.nc')
+        self._partition_scrip_file('mpaso.scrip.nc')
+
+        map_filename = \
+            f'map_{in_mesh_name}_to_{out_mesh_name}_mbtraave.nc'
+
+        self._create_mali_to_mpaso_weights(map_filename)
+
+        self._modify_mali_topo()
+
+        self._remap_mali_to_mpaso(map_filename)
+
+    def _modify_mali_topo(self):
+        """
+        Modify MALI topography to have desired fields and names
+        """
+        logger = self.logger
+        config = self.config
+        logger.info('Modifying MALI topography fields and names')
+
+        ds_mali = xr.open_dataset('mali_topography_orig.nc')
+        if 'Time' in ds_mali.dims:
+            ds_mali = ds_mali.isel(Time=0)
+        bed = ds_mali.bedTopography
+        thickness = ds_mali.thickness
+
+        ice_density = config.getfloat('remap_topography', 'ice_density')
+        ocean_density = config.getfloat('remap_topography', 'ocean_density')
```

Comment:
Check that config section has option `ocean_density` in `setup` since this is not included in all mesh cfg files?

Reply:
I added a check in `setup()` as requested and I set-up (but did not run) the SORRMr4 tests to make sure the new check passes.